packages/core/src/components/Search/Search.tsx

```diff
@@ -1,124 +1,127 @@
-  disabled?: boolean;
-  debounceRate?: number;
-  onBlur?: (event: React.FocusEvent) => void;
-  onFocus?: (event: React.FocusEvent) => void;
-  wrapperClassName?: string;
-  setRef?: () => void;
-  autoComplete?: string;
-  /* BASE_SIZES is exposed on the component itself */
-  size?: (typeof BASE_SIZES)[keyof typeof BASE_SIZES];
-  /* TYPES is exposed on the component itself */
-  type?: SearchType;
-  validation?:
-    | {
-        status: "error" | "success";
-        text: string;
-      }
-    | { text: string };
-  inputAriaLabel?: string;
-  searchResultsContainerId?: string;
-  activeDescendant?: string;
-  /*  Icon names labels for a11y */
-  iconNames?: {
-    layout: string;
-    primary: string;
-    secondary: string;
-  };
-  /** shows loading animation */
-  loading?: boolean;
-}
-
-const Search: VibeComponent<SearchProps, unknown> & {
-  sizes?: typeof BASE_SIZES;
-  types?: typeof SearchType;
-} = forwardRef<unknown, SearchProps>(
+const Search: SearchComponent = forwardRef(
   (
     {
-      secondaryIconName = CloseIcon,
-      iconName = SearchIcon,
-      onChange = NOOP,
-      autoFocus = false,
-      value = "",
-      placeholder = "",
-      disabled = false,
-      debounceRate = 200,
-      onBlur = NOOP,
-      onFocus = NOOP,
-      wrapperClassName = "",
-      setRef = NOOP,
+      searchIconName = SearchIcon,
+      clearIconName = CloseSmallIcon,
+      clearIconLabel = "Clear",
+      additionalActionRender: AdditionalAction,
+      value,
+      placeholder,
+      size = "medium",
+      disabled,
+      loading,
+      autoFocus,
       autoComplete = "off",
-      size = BASE_SIZES.MEDIUM,
-      type = SearchType.SQUARE,
-      className,
-      id = "search",
-      validation = null,
       inputAriaLabel,
-      searchResultsContainerId = "",
-      activeDescendant = "",
-      iconNames = SearchDefaultIconNames,
-      loading = false,
+      debounceRate = 200,
+      searchResultsContainerId,
+      activeDescendant,
+      onChange,
+      onFocus,
+      onBlur,
+      wrapperClassName,
+      className,
+      id,
       "data-testid": dataTestId
     },
     ref
   ) => {
-    const mergedRef = useMergeRef(ref, setRef);
+    const inputRef = useRef<HTMLInputElement>(null);
+    const mergedRef = useMergeRef<HTMLInputElement>(ref, inputRef);
+
+    const { inputValue, onEventChanged, clearValue } = useDebounceEvent({
+      delay: debounceRate,
+      onChange,
+      initialStateValue: value
+    });
+
+    const onClearIconClick = useCallback(() => {
+      if (disabled) {
+        return;
+      }
+
+      inputRef.current?.focus?.();
+      clearValue();
+    }, [disabled, clearValue]);
+
+    const SearchIcon = (
+      <Icon
+        icon={searchIconName}
+        className={styles.icon}
+        clickable={false}
+        iconType={Icon.type.ICON_FONT}
+        iconSize={size === "small" ? "16px" : "20px"}
+        data-testid={getTestId(ComponentDefaultTestId.SEARCH_ICON, id)}
```

Comment:
you mean to `queryAll`?

Reply:
No, you are providing `id` prop to multiple elements in your component, which mean multiple elements in tests will get the same id which is not good.
Plus I'm saying that I don't think you should give it a special test id, just `icon` is enough as it can be targeted already by targeting nested elements